image/docker_schema2_test.go

```diff
@@ -485,7 +485,6 @@ func TestManifestSchema2UpdatedImage(t *testing.T) {
 		assert.NoError(t, err, mime)
 	}
 	for _, mime := range []string{
-		manifest.DockerV2Schema2MediaType, // This indicates a confused caller, not a no-op
```

Comment:
I’m not thrilled with dropping these checks — confused callers should be detected.

Reply:
I was looking at this and there's some inconsistency with docker schema v1 where it does not have this check.. Is there a particular reason for it? Or should I change it to reflect that behavior as well?

Reply:
The trouble in there is that the manifest looks the same for both MIME types, i.e. by the time `manifestSchema1FromManifest` is called, we can’t tell whether it is a bug trying to convert to “the same” MIME type or a generic caller just trying the next MIME type in the list without understanding that the two are really not different. (In particular, see how `determineManifestConversion` works with `dockerImageDestination.SupportedManifestMIMETypes`.)

Overall, I do think this check is valuable to detect mistakes, but not _so_ valuable that it’s worth forcing every generic caller to understand this schema1 idiosyncrasy.

Reply:
Ahh!! okay. Gotcha. Thanks, that comment in the schema1 file makes a lot more sense with this context!